*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/com/android/internal/widget/ScrollingTabContainerView$TabView.smali	2014-05-01 23:28:56.000000000 +0800
--- ./framework.jar.out/smali/com/android/internal/widget/ScrollingTabContainerView$TabView.smali	2014-05-01 23:29:12.000000000 +0800
***************
*** 7,18 ****
  
  
  # annotations
  .annotation system Ldalvik/annotation/EnclosingClass;
      value = Lcom/android/internal/widget/ScrollingTabContainerView;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x2
      name = "TabView"
  .end annotation
  
--- 7,24 ----
  
  
  # annotations
+ .annotation build Landroid/annotation/OppoHook;
+     level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_ACCESS:Landroid/annotation/OppoHook$OppoHookType;
+     note = "Jianhui.Yu@Plf.SDK,2013.10.05: [-private] Modify for oppoStyle Tab"
+     property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+ .end annotation
+ 
  .annotation system Ldalvik/annotation/EnclosingClass;
      value = Lcom/android/internal/widget/ScrollingTabContainerView;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x0
      name = "TabView"
  .end annotation
  
***************
*** 24,30 ****
  
  .field private mTab:Landroid/app/ActionBar$Tab;
  
! .field private mTextView:Landroid/widget/TextView;
  
  .field final synthetic this$0:Lcom/android/internal/widget/ScrollingTabContainerView;
  
--- 30,42 ----
  
  .field private mTab:Landroid/app/ActionBar$Tab;
  
! .field mTextView:Landroid/widget/TextView;
!     .annotation build Landroid/annotation/OppoHook;
!         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_ACCESS:Landroid/annotation/OppoHook$OppoHookType;
!         note = "Jianhui.Yu@Plf.SDK,2013.10.05:[-private] Modify for title marquee"
!         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
!     .end annotation
! .end field
  
  .field final synthetic this$0:Lcom/android/internal/widget/ScrollingTabContainerView;
  
