*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/android/provider/Settings.smali	2014-05-01 23:28:59.000000000 +0800
--- ./framework.jar.out/smali/android/provider/Settings.smali	2014-05-01 23:29:14.000000000 +0800
***************
*** 42,47 ****
--- 42,55 ----
  
  .field public static final ACTION_DISPLAY_SETTINGS:Ljava/lang/String; = "android.settings.DISPLAY_SETTINGS"
  
+ .field public static final ACTION_DREAM_SETTINGS:Ljava/lang/String; = "android.settings.DREAM_SETTINGS"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "QCOM ADD"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->QCOM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
+ 
  .field public static final ACTION_INPUT_METHOD_SETTINGS:Ljava/lang/String; = "android.settings.INPUT_METHOD_SETTINGS"
  
  .field public static final ACTION_INPUT_METHOD_SUBTYPE_SETTINGS:Ljava/lang/String; = "android.settings.INPUT_METHOD_SUBTYPE_SETTINGS"
