*** /home/suda/color2.0/device/temp/android_smali/./secondary-framework.jar.out/smali/android/app/ApplicationPackageManager$ResourceName.smali	2014-05-01 23:29:01.000000000 +0800
--- ./secondary-framework.jar.out/smali/android/app/ApplicationPackageManager$ResourceName.smali	2014-05-01 23:29:18.000000000 +0800
***************
*** 4,15 ****
  
  
  # annotations
  .annotation system Ldalvik/annotation/EnclosingClass;
      value = Landroid/app/ApplicationPackageManager;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x1a
      name = "ResourceName"
  .end annotation
  
--- 4,21 ----
  
  
  # annotations
+ .annotation build Landroid/annotation/OppoHook;
+     level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_ACCESS:Landroid/annotation/OppoHook$OppoHookType;
+     note = "Yaojun.Luo@Plf.SDK : [-private] modify for rom theme"
+     property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+ .end annotation
+ 
  .annotation system Ldalvik/annotation/EnclosingClass;
      value = Landroid/app/ApplicationPackageManager;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x18
      name = "ResourceName"
  .end annotation
  
