*** /home/suda/color2.0/device/temp/android_smali/./secondary-framework.jar.out/smali/android/app/LoadedApk.smali	2014-05-01 23:29:01.000000000 +0800
--- ./secondary-framework.jar.out/smali/android/app/LoadedApk.smali	2014-05-01 23:29:18.000000000 +0800
***************
*** 1753,1773 ****
  .end method
  
  .method public getResources(Landroid/app/ActivityThread;)Landroid/content/res/Resources;
!     .locals 3
      .parameter "mainThread"
  
      .prologue
      iget-object v0, p0, Landroid/app/LoadedApk;->mResources:Landroid/content/res/Resources;
  
      if-nez v0, :cond_0
  
!     iget-object v0, p0, Landroid/app/LoadedApk;->mResDir:Ljava/lang/String;
  
!     const/4 v1, 0x0
  
!     const/4 v2, 0x0
  
!     invoke-virtual {p1, v0, v1, v2, p0}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;ILandroid/content/res/Configuration;Landroid/app/LoadedApk;)Landroid/content/res/Resources;
  
      move-result-object v0
  
--- 1753,1788 ----
  .end method
  
  .method public getResources(Landroid/app/ActivityThread;)Landroid/content/res/Resources;
!     .locals 6
      .parameter "mainThread"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      iget-object v0, p0, Landroid/app/LoadedApk;->mResources:Landroid/content/res/Resources;
  
      if-nez v0, :cond_0
  
!     iget-object v1, p0, Landroid/app/LoadedApk;->mPackageName:Ljava/lang/String;
  
!     iget-object v2, p0, Landroid/app/LoadedApk;->mResDir:Ljava/lang/String;
  
!     const/4 v3, 0x0
  
!     const/4 v4, 0x0
! 
!     iget-object v0, p0, Landroid/app/LoadedApk;->mCompatibilityInfo:Landroid/view/CompatibilityInfoHolder;
! 
!     invoke-virtual {v0}, Landroid/view/CompatibilityInfoHolder;->get()Landroid/content/res/CompatibilityInfo;
! 
!     move-result-object v5
! 
!     move-object v0, p1
! 
!     invoke-virtual/range {v0 .. v5}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;ILandroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
  
      move-result-object v0
  
