*** /home/suda/color2.0/device/temp/android_smali/./secondary-framework.jar.out/smali/android/app/Notification$BigTextStyle.smali	2014-05-01 23:29:00.000000000 +0800
--- ./secondary-framework.jar.out/smali/android/app/Notification$BigTextStyle.smali	2014-05-01 23:29:17.000000000 +0800
***************
*** 42,50 ****
  
  .method private makeBigContentView()Landroid/widget/RemoteViews;
      .locals 8
  
      .prologue
!     const v7, 0x1020330
  
      const/4 v2, 0x0
  
--- 42,55 ----
  
  .method private makeBigContentView()Landroid/widget/RemoteViews;
      .locals 8
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_RESOURCE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jiamiao.He@Plf.Keyguard, 2013.02.04: make notification to oppo\'s"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
!     const v7, 0xc02047b
  
      const/4 v2, 0x0
  
***************
*** 77,83 ****
      #setter for: Landroid/app/Notification$Builder;->mContentText:Ljava/lang/CharSequence;
      invoke-static {v1, v3}, Landroid/app/Notification$Builder;->access$702(Landroid/app/Notification$Builder;Ljava/lang/CharSequence;)Ljava/lang/CharSequence;
  
!     const v1, 0x1090091
  
      invoke-virtual {p0, v1}, Landroid/app/Notification$BigTextStyle;->getStandardView(I)Landroid/widget/RemoteViews;
  
--- 82,88 ----
      #setter for: Landroid/app/Notification$Builder;->mContentText:Ljava/lang/CharSequence;
      invoke-static {v1, v3}, Landroid/app/Notification$Builder;->access$702(Landroid/app/Notification$Builder;Ljava/lang/CharSequence;)Ljava/lang/CharSequence;
  
!     const v1, 0xc09041f
  
      invoke-virtual {p0, v1}, Landroid/app/Notification$BigTextStyle;->getStandardView(I)Landroid/widget/RemoteViews;
  
***************
*** 86,92 ****
      .local v0, contentView:Landroid/widget/RemoteViews;
      if-eqz v6, :cond_0
  
!     const v1, 0x102032f
  
      move v3, v2
  
--- 91,97 ----
      .local v0, contentView:Landroid/widget/RemoteViews;
      if-eqz v6, :cond_0
  
!     const v1, 0xc020477
  
      move v3, v2
  
***************
*** 103,109 ****
  
      invoke-virtual {v0, v7, v2}, Landroid/widget/RemoteViews;->setViewVisibility(II)V
  
!     const v1, 0x1020015
  
      const/16 v2, 0x8
  
--- 108,114 ----
  
      invoke-virtual {v0, v7, v2}, Landroid/widget/RemoteViews;->setViewVisibility(II)V
  
!     const v1, 0xc020476
  
      const/16 v2, 0x8
  
