*** /home/suda/color2.0/device/temp/android_smali/./secondary-framework.jar.out/smali/android/app/Notification.smali	2014-05-01 23:29:00.000000000 +0800
--- ./secondary-framework.jar.out/smali/android/app/Notification.smali	2014-05-01 23:29:18.000000000 +0800
***************
*** 92,97 ****
--- 92,103 ----
  .field public contentView:Landroid/widget/RemoteViews;
  
  .field public contentViewTouchHandle:I
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jiamiao.He@Plf.Keyguard, 2012.06.04:MTK FIELD"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->MTK:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
  
  .field public defaults:I
  
***************
*** 122,129 ****
--- 128,147 ----
  .field public priority:I
  
  .field public simId:J
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jiamiao.He@Plf.Keyguard, 2012.06.04:MTK FIELD"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->MTK:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
  
  .field public simInfoType:I
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jiamiao.He@Plf.Keyguard, 2012.06.04:MTK FIELD"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->MTK:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
  
  .field public sound:Landroid/net/Uri;
  
***************
*** 838,850 ****
      .parameter "contentTitle"
      .parameter "contentText"
      .parameter "contentIntent"
      .annotation runtime Ljava/lang/Deprecated;
      .end annotation
  
      .prologue
!     const v6, 0x1020064
  
!     const v4, 0x1020006
  
      new-instance v0, Landroid/widget/RemoteViews;
  
--- 856,874 ----
      .parameter "contentTitle"
      .parameter "contentText"
      .parameter "contentIntent"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_RESOURCE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jiamiao.He@Plf.Keyguard, 2013.02.04: make notification to oppo\'s"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
      .annotation runtime Ljava/lang/Deprecated;
      .end annotation
  
      .prologue
!     const v6, 0xc020472
  
!     const v4, 0xc020434
  
      new-instance v0, Landroid/widget/RemoteViews;
  
***************
*** 852,858 ****
  
      move-result-object v2
  
!     const v3, 0x109008e
  
      invoke-direct {v0, v2, v3}, Landroid/widget/RemoteViews;-><init>(Ljava/lang/String;I)V
  
--- 876,882 ----
  
      move-result-object v2
  
!     const v3, 0xc09041a
  
      invoke-direct {v0, v2, v3}, Landroid/widget/RemoteViews;-><init>(Ljava/lang/String;I)V
  
***************
*** 878,884 ****
  
      invoke-virtual {v0, v4, v2, v3}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
!     const v2, 0x1020327
  
      const-string v3, "setBackgroundResource"
  
--- 902,908 ----
  
      invoke-virtual {v0, v4, v2, v3}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
!     const v2, 0xc020474
  
      const-string v3, "setBackgroundResource"
  
***************
*** 889,902 ****
      :cond_1
      if-eqz p2, :cond_2
  
!     const v2, 0x1020016
  
      invoke-virtual {v0, v2, p2}, Landroid/widget/RemoteViews;->setTextViewText(ILjava/lang/CharSequence;)V
  
      :cond_2
      if-eqz p3, :cond_3
  
!     const v2, 0x1020046
  
      invoke-virtual {v0, v2, p3}, Landroid/widget/RemoteViews;->setTextViewText(ILjava/lang/CharSequence;)V
  
--- 913,926 ----
      :cond_1
      if-eqz p2, :cond_2
  
!     const v2, 0xc02049c
  
      invoke-virtual {v0, v2, p2}, Landroid/widget/RemoteViews;->setTextViewText(ILjava/lang/CharSequence;)V
  
      :cond_2
      if-eqz p3, :cond_3
  
!     const v2, 0xc020435
  
      invoke-virtual {v0, v2, p3}, Landroid/widget/RemoteViews;->setTextViewText(ILjava/lang/CharSequence;)V
  
***************
*** 929,935 ****
      move-result-object v1
  
      .local v1, f:Ljava/text/NumberFormat;
!     const v2, 0x102032e
  
      iget v3, p0, Landroid/app/Notification;->number:I
  
--- 953,959 ----
      move-result-object v1
  
      .local v1, f:Ljava/text/NumberFormat;
!     const v2, 0xc020473
  
      iget v3, p0, Landroid/app/Notification;->number:I
  
