*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/com/android/internal/os/ZygoteInit.smali	2014-05-01 23:28:56.000000000 +0800
--- ./framework.jar.out/smali/com/android/internal/os/ZygoteInit.smali	2014-05-01 23:29:12.000000000 +0800
***************
*** 1396,1401 ****
--- 1396,1406 ----
  
  .method private static preloadResources()V
      .locals 10
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jianjun.Dan@Plf.SDK,2013.10.05: modify for preload oppo resources"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      invoke-static {}, Ldalvik/system/VMRuntime;->getRuntime()Ldalvik/system/VMRuntime;
***************
*** 1426,1431 ****
--- 1431,1438 ----
  
      invoke-static {v6, v7}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     invoke-static {v3}, Lcom/android/internal/os/ZygoteInit;->preloadOppoResources(Ldalvik/system/VMRuntime;)V
+ 
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v4
***************
*** 2199,2201 ****
--- 2206,2348 ----
      .restart local v11       #parsedArgs:Lcom/android/internal/os/ZygoteConnection$Arguments;
      goto :goto_0
  .end method
+ 
+ .method private static preloadOppoResources(Ldalvik/system/VMRuntime;)V
+     .locals 8
+     .parameter "runtime"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jianjun.Dan@Plf.SDK,2013.10.05: add for preload resource"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v2
+ 
+     .local v2, startTime:J
+     sget-object v4, Lcom/android/internal/os/ZygoteInit;->mResources:Landroid/content/res/Resources;
+ 
+     const v5, 0xc070411
+ 
+     invoke-virtual {v4, v5}, Landroid/content/res/Resources;->obtainTypedArray(I)Landroid/content/res/TypedArray;
+ 
+     move-result-object v1
+ 
+     .local v1, ar:Landroid/content/res/TypedArray;
+     invoke-static {p0, v1}, Lcom/android/internal/os/ZygoteInit;->preloadDrawables(Ldalvik/system/VMRuntime;Landroid/content/res/TypedArray;)I
+ 
+     move-result v0
+ 
+     .local v0, N:I
+     invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
+ 
+     const-string v4, "Zygote"
+ 
+     new-instance v5, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v6, "...preloaded "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, " oppo drawable resources in "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v6
+ 
+     sub-long/2addr v6, v2
+ 
+     invoke-virtual {v5, v6, v7}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, "ms."
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v5
+ 
+     invoke-static {v4, v5}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v2
+ 
+     sget-object v4, Lcom/android/internal/os/ZygoteInit;->mResources:Landroid/content/res/Resources;
+ 
+     const v5, 0xc070412
+ 
+     invoke-virtual {v4, v5}, Landroid/content/res/Resources;->obtainTypedArray(I)Landroid/content/res/TypedArray;
+ 
+     move-result-object v1
+ 
+     invoke-static {p0, v1}, Lcom/android/internal/os/ZygoteInit;->preloadColorStateLists(Ldalvik/system/VMRuntime;Landroid/content/res/TypedArray;)I
+ 
+     move-result v0
+ 
+     invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
+ 
+     const-string v4, "Zygote"
+ 
+     new-instance v5, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v6, "...preloaded "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, " oppo color resources in "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v6
+ 
+     sub-long/2addr v6, v2
+ 
+     invoke-virtual {v5, v6, v7}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     const-string v6, "ms."
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v5
+ 
+     invoke-static {v4, v5}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     return-void
+ .end method
