*** /home/suda/color2.0/device/temp/android_smali/./secondary-framework.jar.out/smali/android/widget/SuggestionsAdapter.smali	2014-05-01 23:29:03.000000000 +0800
--- ./secondary-framework.jar.out/smali/android/widget/SuggestionsAdapter.smali	2014-05-01 23:29:21.000000000 +0800
***************
*** 277,282 ****
--- 277,287 ----
  .method private getActivityIcon(Landroid/content/ComponentName;)Landroid/graphics/drawable/Drawable;
      .locals 10
      .parameter "component"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Prf.SDK,2013.10.07: modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      const/4 v6, 0x0
***************
*** 338,346 ****
      move-result-object v4
  
      .local v4, pkg:Ljava/lang/String;
!     iget-object v7, v0, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
! 
!     invoke-virtual {v5, v4, v3, v7}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
--- 343,349 ----
      move-result-object v4
  
      .local v4, pkg:Ljava/lang/String;
!     invoke-virtual {v0, v5}, Landroid/content/pm/ActivityInfo;->loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
