*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/android/content/Context.smali	2014-05-01 23:28:57.000000000 +0800
--- ./framework.jar.out/smali/android/content/Context.smali	2014-05-01 23:29:12.000000000 +0800
***************
*** 44,49 ****
--- 44,57 ----
  
  .field public static final CLIPBOARD_SERVICE:Ljava/lang/String; = "clipboard"
  
+ .field public static final CONNECTIVITY_OPPO_SERVICE:Ljava/lang/String; = "oppo_connectivity"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Congwei.Yan@Plf.CommSrv : Add for connectivity service of OPPO"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
+ 
  .field public static final CONNECTIVITY_SERVICE:Ljava/lang/String; = "connectivity"
  
  .field public static final CONTEXT_IGNORE_SECURITY:I = 0x2
***************
*** 126,131 ****
--- 134,147 ----
  
  .field public static final STORAGE_SERVICE:Ljava/lang/String; = "storage"
  
+ .field public static final TELEPHONY_OPPO_SERVICE:Ljava/lang/String; = "oppo_phone"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Congwei.Yan@Plf.CommSrv : Add for telephony service of OPPO"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
+ 
  .field public static final TELEPHONY_SERVICE:Ljava/lang/String; = "phone"
  
  .field public static final TEXT_SERVICES_MANAGER_SERVICE:Ljava/lang/String; = "textservices"
***************
*** 729,731 ****
--- 745,781 ----
  
  .method public abstract unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  .end method
+ 
+ .method public isOppoStyle()Z
+     .locals 4
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jianjun.Dan@Plf.SDK : add judged oppostyle"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     invoke-virtual {p0}, Landroid/content/Context;->getTheme()Landroid/content/res/Resources$Theme;
+ 
+     move-result-object v2
+ 
+     sget-object v3, Lcom/oppo/internal/R$styleable;->OppoTheme:[I
+ 
+     invoke-virtual {v2, v3}, Landroid/content/res/Resources$Theme;->obtainStyledAttributes([I)Landroid/content/res/TypedArray;
+ 
+     move-result-object v1
+ 
+     .local v1, typedArray:Landroid/content/res/TypedArray;
+     const/4 v2, 0x1
+ 
+     const/4 v3, 0x0
+ 
+     invoke-virtual {v1, v2, v3}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
+ 
+     move-result v0
+ 
+     .local v0, isOppoTheme:Z
+     invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
+ 
+     return v0
+ .end method
