*** /home/suda/color2.0/device/temp/android_smali/./services.jar.out/smali/com/android/server/wm/WindowState.smali	2014-05-01 23:28:53.000000000 +0800
--- ./services.jar.out/smali/com/android/server/wm/WindowState.smali	2014-05-01 23:29:08.000000000 +0800
***************
*** 3574,3585 ****
--- 3574,3597 ----
      .locals 6
      .parameter "doAnimation"
      .parameter "requestAnim"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "zhangkai add for apklock"
+     .end annotation
  
      .prologue
      const/4 v2, 0x1
  
      const/4 v1, 0x0
  
+     iget-object v3, p0, Lcom/android/server/wm/WindowState;->mAttrs:Landroid/view/WindowManager$LayoutParams;
+ 
+     iget-object v4, p0, Lcom/android/server/wm/WindowState;->mChildWindows:Ljava/util/ArrayList;
+ 
+     invoke-static {v3, v4, p1, p2}, Lcom/android/server/wm/OppoWindowStateHelper;->hideLwExtr(Landroid/view/WindowManager$LayoutParams;Ljava/util/ArrayList;ZZ)Z
+ 
+     move-result p2
+ 
      if-eqz p1, :cond_0
  
      iget-object v3, p0, Lcom/android/server/wm/WindowState;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 5133,5138 ****
--- 5145,5154 ----
      .locals 4
      .parameter "doAnimation"
      .parameter "requestAnim"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "zhangkai add for apklock"
+     .end annotation
  
      .prologue
      const/4 v0, 0x0
***************
*** 5221,5227 ****
      if-nez v2, :cond_0
  
      :cond_2
!     sget-boolean v0, Lcom/android/server/wm/WindowManagerService;->DEBUG_VISIBILITY:Z
  
      if-eqz v0, :cond_3
  
--- 5237,5251 ----
      if-nez v2, :cond_0
  
      :cond_2
!     iget-object v0, p0, Lcom/android/server/wm/WindowState;->mAttrs:Landroid/view/WindowManager$LayoutParams;
! 
!     iget-object v2, p0, Lcom/android/server/wm/WindowState;->mChildWindows:Ljava/util/ArrayList;
! 
!     invoke-static {v0, v2, p1, p2}, Lcom/android/server/wm/OppoWindowStateHelper;->showLwExtr(Landroid/view/WindowManager$LayoutParams;Ljava/util/ArrayList;ZZ)Z
! 
!     move-result p2
! 
!     sget-boolean v0, Lcom/android/server/wm/WindowState;->DEBUG_VISIBILITY:Z
  
      if-eqz v0, :cond_3
  
***************
*** 5461,5463 ****
--- 5485,5511 ----
  
      goto :goto_0
  .end method
+ 
+ .method public bridge synthetic getAttachedWindow()Landroid/view/WindowManagerPolicy$WindowState;
+     .locals 1
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/server/wm/WindowState;->getAttachedWindow()Lcom/android/server/wm/WindowState;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method public getAttachedWindow()Lcom/android/server/wm/WindowState;
+     .locals 1
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "zhangkai add for apklock"
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/wm/WindowState;->mAttachedWindow:Lcom/android/server/wm/WindowState;
+ 
+     return-object v0
+ .end method
