*** /home/suda/color2.0/device/temp/android_smali/./services.jar.out/smali/com/android/server/wm/WindowManagerService$H.smali	2014-05-01 23:28:53.000000000 +0800
--- ./services.jar.out/smali/com/android/server/wm/WindowManagerService$H.smali	2014-05-01 23:29:08.000000000 +0800
***************
*** 360,366 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->notifyFocusChanged()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$300(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_7
      if-eqz v19, :cond_1
--- 360,366 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->notifyFocusChanged()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$200(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_7
      if-eqz v19, :cond_1
***************
*** 527,533 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
      monitor-exit v3
  
--- 527,533 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$300(Lcom/android/server/wm/WindowManagerService;)V
  
      monitor-exit v3
  
***************
*** 1569,1575 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
      monitor-exit v3
      :try_end_15
--- 1569,1575 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$300(Lcom/android/server/wm/WindowManagerService;)V
  
      monitor-exit v3
      :try_end_15
***************
*** 1660,1666 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_1a
      monitor-exit v3
--- 1660,1666 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$300(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_1a
      monitor-exit v3
***************
*** 2093,2099 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->stopFreezingDisplayLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$500(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_20
      monitor-exit v3
--- 2093,2099 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->stopFreezingDisplayLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_20
      monitor-exit v3
***************
*** 2130,2136 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #getter for: Lcom/android/server/wm/WindowManagerService;->mWindowsChanged:Z
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$600(Lcom/android/server/wm/WindowManagerService;)Z
  
      move-result v2
  
--- 2130,2136 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #getter for: Lcom/android/server/wm/WindowManagerService;->mWindowsChanged:Z
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$500(Lcom/android/server/wm/WindowManagerService;)Z
  
      move-result v2
  
***************
*** 2152,2158 ****
      const/4 v4, 0x0
  
      #setter for: Lcom/android/server/wm/WindowManagerService;->mWindowsChanged:Z
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$602(Lcom/android/server/wm/WindowManagerService;Z)Z
  
      monitor-exit v3
      :try_end_28
--- 2152,2158 ----
      const/4 v4, 0x0
  
      #setter for: Lcom/android/server/wm/WindowManagerService;->mWindowsChanged:Z
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$502(Lcom/android/server/wm/WindowManagerService;Z)Z
  
      monitor-exit v3
      :try_end_28
***************
*** 2163,2169 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->notifyWindowsChanged()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$700(Lcom/android/server/wm/WindowManagerService;)V
  
      goto/16 :goto_0
  
--- 2163,2169 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->notifyWindowsChanged()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$600(Lcom/android/server/wm/WindowManagerService;)V
  
      goto/16 :goto_0
  
***************
*** 2505,2511 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->copyAnimToLayoutParamsLocked()Z
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$800(Lcom/android/server/wm/WindowManagerService;)Z
  
      move-result v2
  
--- 2505,2511 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->copyAnimToLayoutParamsLocked()Z
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$700(Lcom/android/server/wm/WindowManagerService;)Z
  
      move-result v2
  
***************
*** 2526,2532 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_26
      monitor-exit v3
--- 2526,2532 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
!     invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$300(Lcom/android/server/wm/WindowManagerService;)V
  
      :cond_26
      monitor-exit v3
***************
*** 2556,2562 ****
      iget v4, v0, Landroid/os/Message;->arg2:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->showStrictModeViolation(II)V
!     invoke-static {v2, v3, v4}, Lcom/android/server/wm/WindowManagerService;->access$900(Lcom/android/server/wm/WindowManagerService;II)V
  
      goto/16 :goto_0
  
--- 2556,2562 ----
      iget v4, v0, Landroid/os/Message;->arg2:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->showStrictModeViolation(II)V
!     invoke-static {v2, v3, v4}, Lcom/android/server/wm/WindowManagerService;->access$800(Lcom/android/server/wm/WindowManagerService;II)V
  
      goto/16 :goto_0
  
***************
*** 2645,2651 ****
      move/from16 v0, v27
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyRotationChanged(II)V
!     invoke-static {v2, v14, v0}, Lcom/android/server/wm/WindowManagerService;->access$1000(Lcom/android/server/wm/WindowManagerService;II)V
  
      goto/16 :goto_0
  
--- 2645,2651 ----
      move/from16 v0, v27
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyRotationChanged(II)V
!     invoke-static {v2, v14, v0}, Lcom/android/server/wm/WindowManagerService;->access$900(Lcom/android/server/wm/WindowManagerService;II)V
  
      goto/16 :goto_0
  
***************
*** 2677,2683 ****
      move-object/from16 v1, v18
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyWindowTranstion(ILandroid/view/WindowInfo;)V
!     invoke-static {v2, v0, v1}, Lcom/android/server/wm/WindowManagerService;->access$1100(Lcom/android/server/wm/WindowManagerService;ILandroid/view/WindowInfo;)V
  
      goto/16 :goto_0
  
--- 2677,2683 ----
      move-object/from16 v1, v18
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyWindowTranstion(ILandroid/view/WindowInfo;)V
!     invoke-static {v2, v0, v1}, Lcom/android/server/wm/WindowManagerService;->access$1000(Lcom/android/server/wm/WindowManagerService;ILandroid/view/WindowInfo;)V
  
      goto/16 :goto_0
  
***************
*** 2719,2725 ****
      move/from16 v1, v17
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyRectangleOnScreenRequested(ILandroid/graphics/Rect;Z)V
!     invoke-static {v2, v14, v0, v1}, Lcom/android/server/wm/WindowManagerService;->access$1200(Lcom/android/server/wm/WindowManagerService;ILandroid/graphics/Rect;Z)V
  
      goto/16 :goto_0
  
--- 2719,2725 ----
      move/from16 v1, v17
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyRectangleOnScreenRequested(ILandroid/graphics/Rect;Z)V
!     invoke-static {v2, v14, v0, v1}, Lcom/android/server/wm/WindowManagerService;->access$1100(Lcom/android/server/wm/WindowManagerService;ILandroid/graphics/Rect;Z)V
  
      goto/16 :goto_0
  
***************
*** 2744,2750 ****
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyWindowLayersChanged(Lcom/android/server/wm/DisplayContent;)V
!     invoke-static {v2, v13}, Lcom/android/server/wm/WindowManagerService;->access$1300(Lcom/android/server/wm/WindowManagerService;Lcom/android/server/wm/DisplayContent;)V
  
      goto/16 :goto_0
  
--- 2744,2750 ----
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleNotifyWindowLayersChanged(Lcom/android/server/wm/DisplayContent;)V
!     invoke-static {v2, v13}, Lcom/android/server/wm/WindowManagerService;->access$1200(Lcom/android/server/wm/WindowManagerService;Lcom/android/server/wm/DisplayContent;)V
  
      goto/16 :goto_0
  
***************
*** 2768,2774 ****
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayAddedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1400(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
--- 2768,2774 ----
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayAddedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1300(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
***************
*** 2802,2808 ****
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayRemovedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1500(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
--- 2802,2808 ----
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayRemovedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1400(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
***************
*** 2836,2842 ****
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayChangedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1600(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
--- 2836,2842 ----
      iget v4, v0, Landroid/os/Message;->arg1:I
  
      #calls: Lcom/android/server/wm/WindowManagerService;->handleDisplayChangedLocked(I)V
!     invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$1500(Lcom/android/server/wm/WindowManagerService;I)V
  
      monitor-exit v3
  
