*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/android/content/pm/PackageItemInfo.smali	2014-05-01 23:28:57.000000000 +0800
--- ./framework.jar.out/smali/android/content/pm/PackageItemInfo.smali	2014-05-01 23:29:12.000000000 +0800
***************
*** 11,16 ****
--- 11,26 ----
  .end annotation
  
  
+ # static fields
+ .field private static final TAG:Ljava/lang/String; = "PackageItemInfo"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : add for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
+ 
+ 
  # instance fields
  .field public icon:I
  
***************
*** 342,349 ****
  .end method
  
  .method public loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
!     .locals 4
      .parameter "pm"
  
      .prologue
      iget v1, p0, Landroid/content/pm/PackageItemInfo;->icon:I
--- 352,364 ----
  .end method
  
  .method public loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
!     .locals 5
      .parameter "pm"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Xiaokang.Feng@Plf.SDK : Modify for we can make ConvertIcon Useless"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      iget v1, p0, Landroid/content/pm/PackageItemInfo;->icon:I
***************
*** 358,364 ****
  
      move-result-object v3
  
!     invoke-virtual {p1, v1, v2, v3}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v0
  
--- 373,381 ----
  
      move-result-object v3
  
!     const/4 v4, 0x0
! 
!     invoke-static {p1, v1, v2, v3, v4}, Landroid/app/OppoThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v0
  
***************
*** 548,550 ****
--- 565,634 ----
  
      return-void
  .end method
+ 
+ .method public loadIcon(Landroid/content/pm/PackageManager;Z)Landroid/graphics/drawable/Drawable;
+     .locals 5
+     .parameter "pm"
+     .parameter "isConvertEnable"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "YongDong.Zhang@Plf.SDK : Add for we can make ConvertIcon Useless"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     iget v1, p0, Landroid/content/pm/PackageItemInfo;->icon:I
+ 
+     if-eqz v1, :cond_3
+ 
+     const/4 v0, 0x0
+ 
+     .local v0, dr:Landroid/graphics/drawable/Drawable;
+     if-eqz p2, :cond_0
+ 
+     iget-object v1, p0, Landroid/content/pm/PackageItemInfo;->packageName:Ljava/lang/String;
+ 
+     iget v2, p0, Landroid/content/pm/PackageItemInfo;->icon:I
+ 
+     invoke-virtual {p0}, Landroid/content/pm/PackageItemInfo;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
+ 
+     move-result-object v3
+ 
+     iget-object v4, p0, Landroid/content/pm/PackageItemInfo;->name:Ljava/lang/String;
+ 
+     invoke-static {p1, v1, v2, v3, v4}, Landroid/app/OppoThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     :cond_0
+     if-eqz v0, :cond_1
+ 
+     if-nez p2, :cond_2
+ 
+     :cond_1
+     iget-object v1, p0, Landroid/content/pm/PackageItemInfo;->packageName:Ljava/lang/String;
+ 
+     iget v2, p0, Landroid/content/pm/PackageItemInfo;->icon:I
+ 
+     invoke-virtual {p0}, Landroid/content/pm/PackageItemInfo;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
+ 
+     move-result-object v3
+ 
+     invoke-virtual {p1, v1, v2, v3}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     :cond_2
+     if-eqz v0, :cond_3
+ 
+     .end local v0           #dr:Landroid/graphics/drawable/Drawable;
+     :goto_0
+     return-object v0
+ 
+     :cond_3
+     invoke-virtual {p0, p1}, Landroid/content/pm/PackageItemInfo;->loadDefaultIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     goto :goto_0
+ .end method
