*** /home/suda/color2.0/device/temp/android_smali/./framework.jar.out/smali/android/content/pm/ResolveInfo.smali	2014-05-01 23:28:57.000000000 +0800
--- ./framework.jar.out/smali/android/content/pm/ResolveInfo.smali	2014-05-01 23:29:12.000000000 +0800
***************
*** 15,21 ****
  
  
  # static fields
! .field public static final CREATOR:Landroid/os/Parcelable$Creator;
      .annotation system Ldalvik/annotation/Signature;
          value = {
              "Landroid/os/Parcelable$Creator",
--- 15,21 ----
  
  
  # static fields
! .field public static final CREATOR:Landroid/os/Parcelable$Creator; = null
      .annotation system Ldalvik/annotation/Signature;
          value = {
              "Landroid/os/Parcelable$Creator",
***************
*** 26,31 ****
--- 26,39 ----
      .end annotation
  .end field
  
+ .field private static final TAG:Ljava/lang/String; = "ResolveInfo"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_FIELD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ .end field
+ 
  
  # instance fields
  .field public activityInfo:Landroid/content/pm/ActivityInfo;
***************
*** 694,701 ****
  .end method
  
  .method public loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
!     .locals 6
      .parameter "pm"
  
      .prologue
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
--- 702,714 ----
  .end method
  
  .method public loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
!     .locals 5
      .parameter "pm"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
***************
*** 708,718 ****
  
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
  
!     iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
! 
!     const/4 v5, 0x0
  
!     invoke-virtual {p1, v3, v4, v5}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 721,729 ----
  
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
  
!     const/4 v4, 0x0
  
!     invoke-direct {p0, p1, v3, v4}, Landroid/content/pm/ResolveInfo;->getThemeIcon(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
***************
*** 742,750 ****
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
! 
!     invoke-virtual {p1, v3, v4, v0}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 753,759 ----
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     invoke-direct {p0, p1, v3, v0}, Landroid/content/pm/ResolveInfo;->getThemeIcon(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
***************
*** 1089,1091 ****
--- 1098,1182 ----
  
      goto :goto_2
  .end method
+ 
+ .method private getThemeIcon(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+     .locals 4
+     .parameter "pm"
+     .parameter "packageName"
+     .parameter "ai"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     const/4 v3, 0x0
+ 
+     iget-object v1, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
+ 
+     invoke-direct {p0, p1, v1, v3}, Landroid/content/pm/ResolveInfo;->loadDrawableFromTheme(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     .local v0, dr:Landroid/graphics/drawable/Drawable;
+     if-nez v0, :cond_0
+ 
+     iget-object v1, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
+ 
+     iget v2, p0, Landroid/content/pm/ResolveInfo;->icon:I
+ 
+     invoke-virtual {p1, v1, v2, v3}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     :cond_0
+     return-object v0
+ .end method
+ 
+ .method private loadDrawableFromTheme(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+     .locals 6
+     .parameter "pm"
+     .parameter "packageName"
+     .parameter "ai"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Yaojun.Luo@Plf.SDK : modify for rom theme"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v4, p0, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     .local v4, ci:Landroid/content/pm/ComponentInfo;
+     :goto_0
+     iget v2, p0, Landroid/content/pm/ResolveInfo;->icon:I
+ 
+     iget-object v0, p0, Landroid/content/pm/ResolveInfo;->filter:Landroid/content/IntentFilter;
+ 
+     invoke-static {v0}, Landroid/app/OppoThemeHelper;->isCustomizedIcon(Landroid/content/IntentFilter;)Z
+ 
+     move-result v5
+ 
+     move-object v0, p1
+ 
+     move-object v1, p2
+ 
+     move-object v3, p3
+ 
+     invoke-static/range {v0 .. v5}, Landroid/app/OppoThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;Landroid/content/pm/PackageItemInfo;Z)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ 
+     .end local v4           #ci:Landroid/content/pm/ComponentInfo;
+     :cond_0
+     iget-object v4, p0, Landroid/content/pm/ResolveInfo;->serviceInfo:Landroid/content/pm/ServiceInfo;
+ 
+     goto :goto_0
+ .end method
