*** /home/suda/color2.0/device/temp/android_smali/./services.jar.out/smali/com/android/server/ServerThread.smali	2014-05-01 23:28:52.000000000 +0800
--- ./services.jar.out/smali/com/android/server/ServerThread.smali	2014-05-01 23:29:06.000000000 +0800
***************
*** 178,183 ****
--- 178,188 ----
  
  .method public run()V
      .locals 164
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->CHANGE_CODE:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jun.Zhang@Plf.Framework, to use oppo button light, three pointers move shot screen; TongXi.Li@Plf.Framework modify for oppo_usb"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
  
      .prologue
      const/16 v7, 0xbc2
***************
*** 828,836 ****
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v6, Lcom/android/server/LightsService;
  
!     invoke-direct {v6, v5}, Lcom/android/server/LightsService;-><init>(Landroid/content/Context;)V
      :try_end_a
      .catch Ljava/lang/RuntimeException; {:try_start_a .. :try_end_a} :catch_1
  
--- 833,841 ----
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v6, Lcom/android/server/OppoLightsService;
  
!     invoke-direct {v6, v5}, Lcom/android/server/OppoLightsService;-><init>(Landroid/content/Context;)V
      :try_end_a
      .catch Ljava/lang/RuntimeException; {:try_start_a .. :try_end_a} :catch_1
  
***************
*** 862,872 ****
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v154, Lcom/android/server/VibratorService;
  
      move-object/from16 v0, v154
  
!     invoke-direct {v0, v5}, Lcom/android/server/VibratorService;-><init>(Landroid/content/Context;)V
      :try_end_c
      .catch Ljava/lang/RuntimeException; {:try_start_c .. :try_end_c} :catch_5f
  
--- 867,877 ----
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v154, Lcom/android/server/OppoVibratorService;
  
      move-object/from16 v0, v154
  
!     invoke-direct {v0, v5}, Lcom/android/server/OppoVibratorService;-><init>(Landroid/content/Context;)V
      :try_end_c
      .catch Ljava/lang/RuntimeException; {:try_start_c .. :try_end_c} :catch_5f
  
***************
*** 2325,2330 ****
--- 2330,2341 ----
  
      :cond_b
      :goto_37
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, v162
+ 
+     invoke-direct {v0, v5, v1}, Lcom/android/server/ServerThread;->initOppoExService(Landroid/content/Context;Lcom/android/server/wm/WindowManagerService;)V
+ 
      const/16 v99, 0x0
  
      .local v99, epoMgr:Lcom/mediatek/common/epo/IMtkEpoClientManager;
***************
*** 5587,5589 ****
--- 5598,5646 ----
      .restart local v25       #networkPolicy:Lcom/android/server/net/NetworkPolicyManagerService;
      goto/16 :goto_40
  .end method
+ 
+ .method private initOppoExService(Landroid/content/Context;Lcom/android/server/wm/WindowManagerService;)V
+     .locals 4
+     .parameter "context"
+     .parameter "wm"
+     .annotation build Landroid/annotation/OppoHook;
+         level = .enum Landroid/annotation/OppoHook$OppoHookType;->NEW_METHOD:Landroid/annotation/OppoHook$OppoHookType;
+         note = "Jun.Zhang@Plf.Framework, for start oppo ex service, three pointers move shot screen, Wanglan add for smali"
+         property = .enum Landroid/annotation/OppoHook$OppoRomType;->ROM:Landroid/annotation/OppoHook$OppoRomType;
+     .end annotation
+ 
+     .prologue
+     :try_start_0
+     const-string v2, "SystemServer"
+ 
+     const-string v3, "Oppo Expand Service"
+ 
+     invoke-static {v2, v3}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     new-instance v1, Lcom/android/server/oppo/OppoExService;
+ 
+     invoke-direct {v1, p1, p2}, Lcom/android/server/oppo/OppoExService;-><init>(Landroid/content/Context;Lcom/android/server/wm/WindowManagerService;)V
+ 
+     .local v1, oppoExService:Lcom/android/server/oppo/OppoExService;
+     const-string v2, "OPPOExService"
+ 
+     invoke-static {v2, v1}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
+     :try_end_0
+     .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
+ 
+     .end local v1           #oppoExService:Lcom/android/server/oppo/OppoExService;
+     :goto_0
+     return-void
+ 
+     :catch_0
+     move-exception v0
+ 
+     .local v0, e:Ljava/lang/Throwable;
+     const-string v2, "SystemServer"
+ 
+     const-string v3, "Failure starting Oppo Service"
+ 
+     invoke-static {v2, v3, v0}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+ 
+     goto :goto_0
+ .end method
